src/AcrolinxEndpoint.php

```diff
@@ -182,19 +182,34 @@ public function getCapabilities(string $authToken): PromiseInterface
      * @param CheckRequest $request
      * @return PromiseInterface containing {@see \Acrolinx\SDK\Models\CheckResponse} or Exception
      */
-    public function check(string $authToken, CheckRequest $request): PromiseInterface
+    public function check(string $authToken, CheckRequest $request, $attempt = 1): PromiseInterface
     {
         $deferred = new Deferred();
 
         $this->client->post($this->props->platformUrl . '/api/v1/checking/checks',
             $this->getCommonHeaders($authToken), $request->getJson())->then(function (ResponseInterface $response)
-        use ($deferred) {
+        use ($request, $authToken, $attempt, $deferred) {
             $checkResponse = new CheckResponse($response);
             $deferred->resolve($checkResponse);
-        }, function (Exception $reason) use ($deferred) {
-            $exception = new AcrolinxServerException($reason->getMessage(), $reason->getCode(),
-                $reason->getPrevious(), 'Submitting check failed');
-            $deferred->reject($exception);
+        }, function (Exception $reason) use ($request, $authToken, $attempt, $deferred) {
+            $responseCode = $reason->getCode();
+            $retryAfterExists = $reason->getResponse()->getHeaders()['retry-after'];
```

Comment:
Use case insensitive matching. 

The header name could be "Retry-After" or "RETRY-AFTER" or any case, all are valid headers

Reply:
Updated. Please take a look

Reply:
@loaiabdelhalim Tests seem to be failing now...